sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Security/Ucfg/EntryPointRecognizer_IsEntryPoint.cs

```diff
@@ -0,0 +1,128 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+extern alias csharp;
+using csharp::SonarAnalyzer.Security.Ucfg;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace SonarAnalyzer.UnitTest.Security.Ucfg
+{
+    [TestClass]
+    public class EntryPointRecognizer_IsEntryPoint : UcfgBuilderTestBase
+    {
+        [TestMethod]
+        public void Public_Controller_Methods_Are_EntryPoints()
```

Comment:
It would be nice to add an extra test with inner classes as you are checking the effective accessibility. For example a private class with a public method.

Reply:
However, I would need to see if inner classes could be controllers at all...

Reply:
If they can't then the `GetEffectiveAccessibility` is not needed :)

Reply:
`GetEffectiveAccessibility` is needed because it is used to combine the accessibility of the method and its containing class.

Reply:
I shouldn't have reviewed at this time yesterday v_v